### ISSUE TITLE
SMS support in OpenSRP

### ISSUE THREAD

**Issue by maimoonak** [OpenSRP/opensrp-server#14]
SMS Notifier (Integration with external SMS sender API/Rapid pro server). Could it be elaborated more. What are the requirements, what min do we need and what max we have in plan for final product.

I would also suggest to design the component generic so that same module can handle email, and MMS content too.

@mberg , @djazayeri , @alihabib , @ndisha , @raihan-mpower , @julkarnain , @sohelsarder , @jayantchaudhari7 , @gigix , @cagulas , @dimasciput , @raufdesu , @kercobo , @wahid-nwr

**Comment by ndisha** [comment]
@maimoonak I agree with ensuring that we have the component handle email & MMS as well.
Looking at the possibilities of setting up RapidPro as discussed today.

@mberg @cagulas

**Comment by ndisha** [comment]
Ona has finished setting up a Rapidpro instance - this can be used to explore SMS integration into OpenSRP. @sohelsarder @koros we need to talk about this and probably have some work done and show cased on the 9th?

**Comment by sohelsarder** [comment]
Sure, look forward to work on it and together we need to finalize way forward of common notifier.

**Comment by ndisha** [comment]
@sohelsarder @numeranazneen I know the SMS integration needs in Bangladesh are pretty clear, so my request to you is can we get use cases and the different scenarios for which we would use SMS in your OpenSRP implementation?

**Comment by numeranazneen** [comment]
The SMS requirement is already mentioned in the Service Schedule doc and uploaded in Google Drive. Please check the link below (starting from row 36)-

https://docs.google.com/spreadsheets/d/1UR6dzNAsDU5M9i_7Uy4XvJ3LJqmxhLhzawvHCOAMVw8/edit#gid=0

**Comment by ndisha** [comment]
Thanks @numeranazneen. Am also adding some of the functionality we discussed in Dhaka that were a Must Have for the sites SMS integration needs:

* [ ] Receive SMS or voice messages informing me of due dates and risk situations and impending FHW visits, or personalized health advice
* [ ] Send SMS or dial a hotline number to register an event or get emergency support - the log becomes part of OpenSRP and can trigger actions and schedules
* [ ] Send SMS reminders to individual clients before visits (this is based on the schedules as can be seen on the schedules sms requirements worksheet above), or in bulk before camps
* [ ] Get notifications and flags from the app or via SMS to respond to emergency situations or get critical messages from supervisors (Low priority for all sites)
* [ ] Receive medical advice/feedback from doctor through SMS/voice message when there is no data network (Low priority for all sites)
